### DISCUSSION TITLE
Gatsby is sourcing READMEs

### DISCUSSION THREAD

**Discussion by fostimus** [gatsbyjs/gatsby#36761] [Help]
I use `gatsby-source-filesystem` to take in markdown files for articles. It's limited by directory, e.g. `content/blog-posts`.

I'm using graphiql during development to see what gets sourced, and Gatsby is processing markdown files from the entire project, like my top-level README and directories I did not configured to be sourced.

Is this normal behavior? I've used the `ignore` option on `gatsby-source-filesystem` for READMEs, but that didn't work. How can I prevent these files from being sourced?

**Comment by LekoArts** [comment]
The sourcing must be set up somewhere, Gatsby doesn't automatically picks those files up. You can run this query to check with plugin and with path is sourcing the files:

```graphql
{
  allFile {
    nodes {
      sourceInstanceName
      absolutePath
      internal {
        owner
      }
    }
  }
}
```

**Reply by fostimus** [reply]
Seems like all files in the project are getting sourced via `gatsby-source-filesystem`. The files we don't have set up to be sourced have `sourceInstanceName` set to `__PROGRAMMATIC__`, what does that value mean?

**Reply by fostimus** [reply]
This is still affecting us. I've tried using the `ignore` option and it still doesn't ignore the README.md files. Tried the following fileglobs: 

 `**/README\\.md`,
          `**/README.md`,
          `README.md`,
          `README\\.md`,
          `**/readme-*\\.md`,
          `**/readme-*.md`,
          `readme-*\\.md`,
          `readme-*.md`,

**Reply by gh640** [reply]
@fostimus 

`__PROGRAMMATIC__` is likely used when the required `name` option is not set.

https://github.com/gatsbyjs/gatsby/blob/89a3232efdf2b69a67ad746294ab750c631f6afd/packages/gatsby-source-filesystem/src/create-file-node.js#L70

Could you share your `gatsby-config.js` or make a minimal reproduction? It usually doesn't occur (if I understand it correctly) and it's difficult for others to help with the provided information.

**Reply by fostimus** [reply]
@gh640 I will try to get a minimal repro up, but likely don't have the time. Will come back here with one.

A hunch (if it helps) is that we use [docz](https://www.npmjs.com/package/gatsby-theme-docz) to generate documentation, which I know uses `gatsby-source-mdx`, and is horribly outdated. I could see a case where docz is grabbing all *.md(x) files in our repo.
